ckanext/oidc_pkce/plugin.py

```diff
@@ -40,10 +46,39 @@ def get_helpers(
     ):
         return helpers.get_helpers()
 
-    if not tk.check_ckan_version("2.10"):
-        p.implements(p.IAuthenticator, inherit=True)
+    # IAuthenticator
+
+    if tk.check_ckan_version("2.10"):
+        def logout(self):
+            """ We want to return a view after the regular logout logic,
+            rather than before.
+
+            We set a flag to indicate that we're in the middle of logout,
+            then call the regular logout view. The view calls a second
+            instance of this function, which detects the flag and no-ops,
+            allowing the view to proceed and wipe the session.
 
-        # IAuthenticator
+            After it completes, we assemble a redirect and pass that back
+            to the code that originally called this function.
+            """
+            if session.pop("_in_logout", False):
+                return None
+            elif not current_user.name:
+                # not logged in, do nothing
+                return None
+            else:
+                log.info("Logging out [%s]", current_user.name)
+                sso_logout_url = config.logout_url()
+                if not sso_logout_url:
+                    log.info("No SSO logout path configured, logout of [%s] will be local only",
+                             current_user.name)
+                    return None
+                session.put("_in_logout", True)
+                original_response = user_view.logout()
+                log.debug("Redirecting [%s] to SSO logout at: %s",
+                          current_user.name, sso_logout_url)
+                return redirect(sso_logout_url + '?redirect_uri=' + original_response.location)
```

Comment:
are we share this won't be globbed by our other plugins security?

Reply:
I'm not sure I follow. What plugin would interfere with this?